proposals/infra/INF-0004-validator-hashing.md

```diff
@@ -195,6 +163,93 @@ multiple-year-old preview compiler for generating final compiled shaders for a
 title. The high rate of bugs in preview compilers contributes to making this
 even less likely.
 
+## Detailed Design
+
+### D3D Runtime Behavior
+
+Starting with AgilitySDK version TBD, the runtime adopts new behavior for how 
+and when the shader hash is validated, including support for bytecode validation 
+in the debug layer.
+
+For shaders with the `PREVIEW_BYPASS` hash, the shader is allowed to execute 
+only if developer mode and the experimental feature 
+`D3D12ExperimentalShaderModels` is enabled, otherwise the runtime will produce
+an error.
+
+For shaders with the `BYPASS` hash, shaders can run without validating the hash 
+regardless of whether the machine is in developer mode as long as the shader 
+targets a supported non-experimental shader model.
+
+All other values of the hash are assumed to be real hash values.
+
+For shaders that contain a real hash value, the behavior is unchanged. The hash
+is verified and the shader executes as it does with existing versions of the 
+runtime.
+
+> Note: an important behavior change here is that shaders with zero'd hash data
+> will always be treated as invalid by the runtime and rejected.
+> 
+> An exception is for a period of time already existing preview shaders that used 
+> zero'd hash data will be treated like having the `PREVIEW_BYPASS` hash in that
+> preview runtimes will allow them to be used, at least until those preview shaders
+> and runtimes are no longer supported.  In contrast to `PREVIEW_BYPASS` shaders,
+> these legacy zero'd has shaders will not run have the validator invoked on them
+> by the runtime/debug layer since that wasn't expected of them.
```

Comment:
I think this section might have been overly edited, I'm not exactly sure what some of it is trying to say.  An attempt at clarifying:

```suggestion
> We have an exception to allow for legacy zero'd hash shaders. These were built
> using earlier previews and so they're treated as if they had the `PREVIEW_BYPASS` hash set,
> except that the validator will not be invoked on them since this wasn't expected at the time
> they were generated. This exception will be removed once runtimes that support these preview 
> shaders are no longer supported.
```